reverse_engineering/helpers/snowflakeHelper.js

```diff
@@ -1136,6 +1149,120 @@ const getMaterializedViewData = async fullName => {
+	const match = query.match(regex);
+
+	if (match) {
+		return match[1] || match[2];
+	}
+}
+
+function getTargetLagStringValue(timeString) {
+	const timePattern = /(\d+)\s*(day|hour|minute|second)s?/g;
+	let totalSeconds = 0;
+	let match;
+	while ((match = timePattern.exec(timeString)) !== null) {
+		const value = parseInt(match[1]);
+		const unit = match[2];
+
+		switch (unit) {
+			case 'day':
+				totalSeconds += value * SECONDS_IN_DAY;
+				break;
+			case 'hour':
+				totalSeconds += value * SECONDS_IN_HOUR;
+				break;
+			case 'minute':
+				totalSeconds += value * SECONDS_IN_MINUTE;
+				break;
+			case 'second':
+				totalSeconds += value;
+				break;
+		}
+	}
+
+	if (totalSeconds >= SECONDS_IN_DAY) {
+		const days = totalSeconds / SECONDS_IN_DAY;
+		return {
+			targetLagAmount: Math.round(days),
+			targetLagTimeSpan: 'days',
+		};
+	} else if (totalSeconds >= SECONDS_IN_HOUR) {
+		const hours = totalSeconds / SECONDS_IN_HOUR;
+		return {
+			targetLagAmount: Math.round(hours),
+			targetLagTimeSpan: 'hours',
+		};
+	} else if (totalSeconds >= SECONDS_IN_MINUTE) {
+		const minutes = totalSeconds / SECONDS_IN_MINUTE;
+		return {
+			targetLagAmount: Math.round(minutes),
+			targetLagTimeSpan: 'minutes',
+		};
+	}
+	return {
+		targetLagAmount: Math.round(totalSeconds),
+		targetLagTimeSpan: 'seconds',
+	};
+}
+
+function getTargetLag(targetLag) {
+	if (targetLag === 'DOWNSTREAM') {
+		return {
+			targetLagDownstream: true,
+		};
+	}
+
+	return getTargetLagStringValue(targetLag);
+}
+
+const getDynamicTableData = async fullName => {
+	const [dbName, schemaName, tableName] = fullName.split('.');
+
+	try {
+		const rows = await execute(
+			`SHOW DYNAMIC TABLES LIKE '${removeQuotes(tableName)}' IN SCHEMA "${removeQuotes(dbName)}"."${removeQuotes(schemaName)}"`,
+		);
+
+		const data = _.first(rows);
+		const refreshMode = _.get(data, 'refresh_mode', '').toLowerCase();
+		const warehouse = _.get(data, 'warehouse', '');
+		const text = _.get(data, 'text', '');
+
+		const targetLag = getTargetLag(_.get(data, 'target_lag', ''));
+		const externalVolume = getOptionValue(text, 'EXTERNAL_VOLUME');
+		const initialize = getOptionValue(text, 'INITIALIZE').toLowerCase();
+		const catalog = getOptionValue(text, 'CATALOG');
+		const baseLocation = getOptionValue(text, 'BASE_LOCATION');
+		const DATA_RETENTION_TIME_IN_DAYS = getOptionValue(text, 'DATA_RETENTION_TIME_IN_DAYS');
+		const MAX_DATA_EXTENSION_TIME_IN_DAYS = getOptionValue(text, 'MAX_DATA_EXTENSION_TIME_IN_DAYS');
+
+		const selectStatement = text
+			.slice(text.indexOf('AS\n') + 4, -1)
+			.split('\n')
+			.filter(Boolean)
+			.map(line => (line.startsWith('\t') ? line.slice(1, -1) : line))
+			.join('\n');
+
+		return {
+			dynamic: true,
+			targetLag,
+			refreshMode,
+			warehouse,
+			selectStatement,
+			externalVolume,
+			catalog,
+			initialize,
+			baseLocation,
+			DATA_RETENTION_TIME_IN_DAYS,
+			MAX_DATA_EXTENSION_TIME_IN_DAYS,
+		};
+	} catch (err) {
+		return {};
```

Comment:
@Nightlngale please make sur to log errors and make them bubble if required.  Just silenting any error doesn't help on the long run :)